src/static/js/modules/youtube.js

```diff
@@ -0,0 +1,177 @@
+  this.$element = $( element );
+  this.options = $.extend( true, {}, this.defaults, options );
+  this.init();
+}
+
+// Youtube methods and shared properties
+Youtube.prototype = {
+
+  constructor: Youtube,
+
+  defaults: {
+     playerOptions : {
+          height: '410',
+          width: '640',
+          videoId: 'en0Iq8II4fA',
+          playerVars: { 
+              'autoplay': 1, 
+              'suggestedQuality': 'highres'
+          },
+          events: {
+              'onReady': $.noop,
+              'onStateChange': $.noop
+          }
+      },
+      playerInitialized : false
+  },
+
+  _bindAll: function() { 
+    var self = this;
+    var proxy = $.proxy;
+
+    // TODO: Swap this out for Function.bind
+    // and Object.keys. 
+    $.each( this, function( key, _function ){
+        if ( $.isFunction( _function ) ) {
+            self[key] = proxy( _function, self );
+        }
+    });
+  },
+
+  _embedScript: function() {
+      window.onYouTubeIframeAPIReady = this.initPlayer;
+
+      var tag = document.createElement('script');
+      tag.src = "http://www.youtube.com/iframe_api";
+      var firstScriptTag = document.getElementsByTagName('script')[0];
+      firstScriptTag.parentNode.insertBefore( tag, firstScriptTag );
+
+      return this;
+  },
+
+  init: function() {
+      this._bindAll();
+      this.initUI();
+
+      var youtubeEvents = this.defaults.playerOptions.events;
+      youtubeEvents.onReady = this.onPlayerReady;
+      youtubeEvents.onStateChange = this.onPlayerStateChange;
+
+      return this;
+  },
+
+  initUI: function() {
+      var ui = this.$element;
+      ui.on( 'click', '.youtube-player_play-btn', this.play );
+      ui.on( 'click', '.youtube-player_close-btn', this.close );
+
+      return this;
+  },
+
+  destroy: function() {
+      var self = this;
+      this.$element.remove();
+
+      $.each( this, function( key, _function ){
+        delete  this[key];
+      }, this );
+      
+      return this;
+  },
+
+  initPlayer: function() {
+      if ( window.YT && window.YT.Player ) {
+          new YT.Player( 'youtube-player', this.defaults.playerOptions );
+          this.playerInitialized = true;
+      } else {
+          this._embedScript();
+      }
+
+      return this;
+  },
+
+  onPlayerStateChange: function( event ) {
+      if ( event.data == YT.PlayerState.ENDED ) {
+          this.$element.removeClass(  'yt-playing' );
+      }
+
+      return this;
+  },
+
+  onPlayerReady: function( params ) {
+      if ( params && params.target ) {
+          this.player = params.target;
+      }
+  },
+
+  play: function( e ) {
+      e.preventDefault();
+
+      if( this.playerInitialized ) {
+          this.player.seekTo( 0 );
+          this.player.playVideo();
+      } else {
+          this.initPlayer();
+      }
+
+      this.$element.addClass( 'yt-playing' );
+      $('html, body').animate( { scrollTop: 0 }, 500 );
+
+      return this;
+  },
+
+  close: function(e) {
```

Comment:
Minor: Run ESLint, pretty sure it'll complain and want this to be `function( e )`